### ISSUE TITLE
Is there a video backend supporting nondestructive encoding?

### ISSUE THREAD

**Issue by milong26** [huggingface/lerobot#1579]
I saved images during recording through not deletng folder `images`. When I try to compare the first frame.png in `images` folder and dataset=make_dataset(config)'s first image, I found the saved png file is nondestructive. But the image I got by lerobot is not.

How I find:
in `def save_episode` 
```
        # img_dir = self.root / "images"
        # if img_dir.is_dir():
        #     shutil.rmtree(self.root / "images")
```
This has been moved in latest version. now:

```
    def encode_episode_videos(self, episode_index: int) -> None:
            ...
            encode_video_frames(img_dir, video_path, self.fps, overwrite=True)
            shutil.rmtree(img_dir)
```

I saved some images through recording with one channel filled with zero. Then read the saved png through cv2, it showed it has a 0-filled channel.

Then I try to check whether I can get the same image through lerobot
so I did this in train.py
```
raw_dataloader = torch.utils.data.DataLoader(
        dataset,
        num_workers=cfg.num_workers,
        batch_size=cfg.batch_size,
        shuffle=False,
        sampler=sampler,
        pin_memory=device.type == "cuda",
        drop_last=False,
    )
image_tensor=peek_batch["observation.images.side_depth"][0]
image_np = (image_tensor * 255).permute(1, 2, 0).cpu().numpy().astype(np.uint8)
```
Sadly,`image_np` is really different from real png, it doesn't have a 0-filled channel, and its average data shows larger.

**Comment by milong26** [comment]
Indeed I found this problem after collecting :( 
I'm considering some solutions:
1, Dismiss the error and go on training. But in evaluation, it does get-observation(real) -> predict -> save_video(error here). How the model learn from error data and evaluate through real dta?
2. re-collect(I hate it ;-;) and save png, using png to train. (need more code) 
3. re-collect and save data into array since I only need 16bit for every pixel, but it needs code too and take large memory

Is there any good solution to fix my problem? Thanks!

**Comment by milong26** [comment]
Finally I saved images and read from them 
If anyone met this problem too can choose this way:
1. keep lerobot not deleting images folder before collecting(As I described in description of the issue)
2. modilfy function `_query_videos` in `lerobot_dataset.py` and its upstream/downstream function. To get the right image from local folder, you need to get the index of image (episode_index+frame index) 
3. Variable `ep_idx` in function  `__getitem__`  which triggers `query_videos` can be seen as episode_index, while query_indices is the frame number of image in all images. Such as in episode0 it is right, in episode1,the query_indices starts from size of spisode0.

**Comment by rancheng** [comment]
we have tried before, I saved the depth tensor to the `state` section, and route it to the `observation` and mask out depth tensor in the `state` while training

**Comment by milong26** [comment]
> we have tried before, I saved the depth tensor to the `state` section, and route it to the `observation` and mask out depth tensor in the `state` while training

@rancheng Save all depth into `state` in one episode? the parquet file will be large :o (Other method has same problem)
